src/hooks/useClickOutside.tsx

```diff
@@ -0,0 +1,26 @@
+import { RefObject, useEffect } from 'react';
+
+const UseClickOutside = function (
+  ref: RefObject<HTMLElement>,
+  onClickOutside: () => void,
+  ignoreDataSetProperty: string,
+) {
+  const handleClickOutside = (event: any) => {
+    if (
```

Comment:
In example that You presented to me function handleClickOutside was wrapped in useCallback. Is there any important reason of doing that? 
I did not wrapped this function and everything is working fine .

Reply:
It has impact on performarnce when ref, ignoreDatasetProperty or method would change passed from some place when useclickOutside would be used on large set of list items during rerenders